src/components/course-settings/add-period.cjsx

```diff
@@ -0,0 +1,85 @@
+React = require 'react'
+BS = require 'react-bootstrap'
+_ = require 'underscore'
+{PeriodActions, PeriodStore} = require '../../flux/period'
+{TutorInput} = require '../tutor-input'
+BindStoreMixin = require '../bind-store-mixin'
+
+
+AddPeriodField = React.createClass
+
+  displayName: 'AddPeriodField'
+  propTypes:
+    courseId: React.PropTypes.string
+    label: React.PropTypes.string.isRequired
+    name:  React.PropTypes.string.isRequired
+    default: React.PropTypes.string
+    onChange:  React.PropTypes.func.isRequired
+    autofocus: React.PropTypes.bool
+
+  componentDidMount: ->
+    @refs.input.focus() if @props.autofocus
+
+  onChange: (value) ->
+    @props.onChange(value)
+
+  render: ->
+    <TutorInput
+      ref="input"
+      label={@props.label}
+      default={@props.default}
+      required={true}
+      onChange={@onChange} />
+
+module.exports = React.createClass
+  displayName: 'AddPeriodLink'
+  propTypes:
+    courseId: React.PropTypes.string.isRequired
+    periods: React.PropTypes.array.isRequired
+
+  mixins: [BindStoreMixin]
+  bindStore: PeriodStore
+  bindEvent: 'create'
+
+  getInitialState: ->
+    warning: ''
+
+  performUpdate: ->
+    name = PeriodStore.validatePeriodName(@state.period_name, @props.periods)
+    if not name.error
```

Comment:
**Minor:** This can be `unless`. or switch the code to be:
```coffee
if name.error
  @setState(warning: name.error)
else
  @refs.overlay.hide()
  PeriodActions.create(@props.courseId, period: {name: @state.period_name})
```